src/convertToObject.js

```diff
@@ -6,7 +6,20 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString
+    .split(';')
+    .filter((item) => item.trim() !== '')
+    .map((item) => item.trim())
+    .reduce((stylesObject, item) => {
+      const pair = item.split(':');
+
+      const key = pair[0].trim();
+      const value = pair[1].trim();
+
+      stylesObject[key] = value;
+
+      return stylesObject;
+    }, {});
 }
 
 module.exports = convertToObject;
```

Comment:
The `convertToObject` function implementation looks correct and adheres to the checklist provided. It uses string and array methods to split, trim, filter, and reduce the CSS declarations into an object without mutating the original string. However, there is an unnecessary chaining of `map` after `filter` on lines 11-12. Since you are already using `trim` in the `filter` method, you can directly destructure and trim the key-value pairs in the `reduce` function to avoid the extra iteration over the array.